at_common_flutter/lib/widgets/custom_toast.dart

```diff
@@ -0,0 +1,49 @@
+
+import 'dart:io';
+
+import 'package:flutter/material.dart';
+
+class CustomToast {
+  /// pass [isError] true to have a red bg, [isSuccess] true to have a green bg.
+  static void show(String message, BuildContext context,
+          {Color? bgColor,
+          Color? textColor,
+          double? toastWidth,
+          double fontSize = 12,
+          int duration = 3,
+          double gravity = 10.0,
+          bool isError = false,
+          bool isSuccess = false}) {
+            assert(!(isError && isSuccess), 'Both isError and isSuccess cannot be true');
+            assert(gravity >= 0, 'gravity cannot be less than 0');            
+              ScaffoldMessenger.of(context).showSnackBar(
```

Comment:
@yahu1031 we also have to assert that the given context has a ScaffoldMessenger ancestor.